internal/planner/mapper/mapper.go

```diff
@@ -41,7 +41,75 @@ const (
 	CommitSelection
 )
 
-// ToSelect converts the given [parser.Select] into a [Select].
+// ToOperation converts the given [request.OperationDefinition] into an [Operation].
+//
+// In the process of doing so it will construct the document map required to access the data
+// yielded by the [Operation].
+func ToOperation(
+	ctx context.Context,
+	store client.Store,
+	operationRequest *request.OperationDefinition,
+) (*Operation, error) {
+	operation := &Operation{
+		DocumentMapping: core.NewDocumentMapping(),
+	}
+
+	for i, s := range operationRequest.Selections {
+		switch t := s.(type) {
+		case *request.CommitSelect:
+			s, err := toCommitSelect(ctx, store, t, i)
+			if err != nil {
+				return nil, err
+			}
+			renderKey := core.RenderKey{Index: i}
+			if t.Alias.HasValue() {
+				renderKey.Key = t.Alias.Value()
+			} else {
+				renderKey.Key = t.Name
+			}
+			operation.CommitSelects = append(operation.CommitSelects, s)
+			operation.DocumentMapping.Add(i, s.Name)
+			operation.DocumentMapping.SetChildAt(i, s.DocumentMapping)
+			operation.DocumentMapping.RenderKeys = append(operation.DocumentMapping.RenderKeys, renderKey)
+
+		case *request.Select:
+			s, err := toSelect(ctx, store, ObjectSelection, i, t, "")
+			if err != nil {
+				return nil, err
+			}
+			renderKey := core.RenderKey{Index: i}
+			if t.Alias.HasValue() {
+				renderKey.Key = t.Alias.Value()
+			} else {
+				renderKey.Key = t.Name
+			}
+			operation.Selects = append(operation.Selects, s)
+			operation.DocumentMapping.Add(i, s.Name)
+			operation.DocumentMapping.SetChildAt(i, s.DocumentMapping)
+			operation.DocumentMapping.RenderKeys = append(operation.DocumentMapping.RenderKeys, renderKey)
+
+		case *request.ObjectMutation:
+			m, err := toMutation(ctx, store, t, i)
+			if err != nil {
+				return nil, err
+			}
+			renderKey := core.RenderKey{Index: i}
+			if t.Alias.HasValue() {
+				renderKey.Key = t.Alias.Value()
+			} else {
+				renderKey.Key = t.Name
+			}
+			operation.Mutations = append(operation.Mutations, m)
+			operation.DocumentMapping.Add(i, m.Name)
+			operation.DocumentMapping.SetChildAt(i, m.DocumentMapping)
+			operation.DocumentMapping.RenderKeys = append(operation.DocumentMapping.RenderKeys, renderKey)
```

Comment:
nitpick: the bodies of all 3 cases are very similar. Is there a way to extract common functionality?

Reply:
I moved some of this to a function and reduced most of the duplicate lines